rcl/src/rcl/arguments.c

```diff
@@ -1263,6 +1369,81 @@ _rcl_parse_remap_rule(
   return ret;
 }
 
+rcl_ret_t
+_rcl_parse_param_rule(
+  const char * arg,
+  rcl_params_t * params)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(arg, RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(params, RCL_RET_INVALID_ARGUMENT);
+
+  rcl_lexer_lookahead2_t lex_lookahead = rcl_get_zero_initialized_lexer_lookahead2();
+
+  rcl_ret_t ret = rcl_lexer_lookahead2_init(&lex_lookahead, arg, params->allocator);
+  if (RCL_RET_OK != ret) {
+    return ret;
+  }
+
+  rcl_lexeme_t lexeme1;
+  rcl_lexeme_t lexeme2;
+  char * node_name = NULL;
+  char * param_name = NULL;
+
+  // Check for optional nodename prefix
+  ret = rcl_lexer_lookahead2_peek2(&lex_lookahead, &lexeme1, &lexeme2);
+  if (RCL_RET_OK != ret) {
+    goto cleanup;
+  }
+
+  if (RCL_LEXEME_TOKEN == lexeme1 && RCL_LEXEME_COLON == lexeme2) {
+    ret = _rcl_parse_nodename_prefix(&lex_lookahead, params->allocator, &node_name);
```

Comment:
Watch out, `_rcl_parse_nodename_prefix` _expects_ those two `lexeme`s. Since this prefix is optional, we must check that those are there before expecting them. Otherwise, you may eat up a token before failing when the colon is not found. This functionality was [already in place](https://github.com/ros2/rcl/blob/41792cb6e80d9d22031e8902d36e33a14e2db100/rcl/src/rcl/arguments.c#L1157).